astropy/utils/tests/test_decorators.py

```diff
@@ -250,6 +250,197 @@ def C(cls):
         assert 'deprecated' in A.C.__doc__
 
 
+def test_deprecated_argument():
+    # Tests the decorator with function, method, staticmethod and classmethod.
+
+    class Test(object):
+
+        @classmethod
+        @deprecated_renamed_argument('clobber', 'overwrite', '1.3')
+        def test1(cls, overwrite):
+            return overwrite
+
+        @staticmethod
+        @deprecated_renamed_argument('clobber', 'overwrite', '1.3')
+        def test2(overwrite):
+            return overwrite
+
+        @deprecated_renamed_argument('clobber', 'overwrite', '1.3')
+        def test3(self, overwrite):
+            return overwrite
+
+    @deprecated_renamed_argument('clobber', 'overwrite', '1.3', relax=False)
+    def test1(overwrite):
+        return overwrite
+
+    for method in [Test().test1, Test().test2, Test().test3, test1]:
+        # As positional argument only
+        assert method(1) == 1
+
+        # As new keyword argument
+        assert method(overwrite=1) == 1
+
+        # Using the deprecated name
+        with catch_warnings(AstropyDeprecationWarning) as w:
+            assert method(clobber=1) == 1
+
+        assert len(w) == 1
+        assert '1.3' in str(w[0].message)
```

Comment:
I've seen both styles and this specific example is copied from another test. I can indent it if you want it.